EspionSpotify.Updater/Update.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Diagnostics;
+using System.Drawing;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Ionic.Zip;
+using MetroFramework.Forms;
+
+namespace EspionSpotify.Updater
+{
+    public partial class Update : MetroForm
+    {
+        public Update(string[] args)
+        {
+
+            InitializeComponent();
+            Thread t = new Thread(() => {
+                ProcessUpdate(args);
+            });
+            t.Start();
+
+        }
+        public void ProcessUpdate(string[] args)
+        {
+            if (args.Length==0)
+            {
+                MessageBox.Show("An error occurred while updating. Please try it again later","Update failed",MessageBoxButtons.OK,MessageBoxIcon.Error);
+                Environment.Exit(1);
+            }
+            string path = args[0];
+            try
+            {
+                extract(path);
+            }
+            catch( Exception ex)
+            {
+                MessageBox.Show("An error occurred while updating. Please try it again later"+ ex.Message, "Update failed", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                Environment.Exit(1);
+            }
+
+            Process p = new Process();
+            p.StartInfo = new ProcessStartInfo();
+            p.StartInfo.FileName= path+"Spytify.exe";
+            p.StartInfo.Arguments = "upgradeSettings";
+            p.Start();
+            Environment.Exit(0);
+        }
+        public void extract(string path)
+        {
+            ZipFile zip = ZipFile.Read(path+"update.zip");
```

Comment:
It may be not there yet?
It also occurs for some users with recorded files, so I don't know if a downloaded file can still be locked / in use by the process that downloaded it.

Reply:
The whole update process is run on a single thread and executed synchronusly. So the extracting will not be started until the downlad was finished and the file exists.